Simplenote/src/main/java/com/automattic/simplenote/di/DataModule.kt

```diff
@@ -0,0 +1,29 @@
+package com.automattic.simplenote.di
+
+import android.content.Context
+import com.automattic.simplenote.Simplenote
+import com.automattic.simplenote.models.Note
+import com.automattic.simplenote.models.Tag
+import com.automattic.simplenote.repositories.SimperiumTagsRepository
+import com.automattic.simplenote.repositories.TagsRepository
+import com.simperium.client.Bucket
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+
+@Module
+@InstallIn(SingletonComponent::class)
+object DataModule {
+
+    @Provides
+    fun providesTagsBucket(@ApplicationContext appContext: Context): Bucket<Tag> = (appContext as Simplenote).tagsBucket
+
+    @Provides
+    fun providesNotesBucket(@ApplicationContext appContext: Context): Bucket<Note> = (appContext as Simplenote).notesBucket
+
+    @Provides
+    fun providesTagsRepository(tagsBucket: Bucket<Tag>, notesBucket: Bucket<Note>): TagsRepository =
+        SimperiumTagsRepository(tagsBucket, notesBucket)
```

Comment:
np, I would change this to use @Binds instead of @Provides, this is because binding will decouple the injection from the constructor's implementation, and if the number of arguments needed for the constructor changes, we won't need to update in the module itself.

A patch to make my suggestion clearer 🙂 
```
Index: Simplenote/src/main/java/com/automattic/simplenote/di/DataModule.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Simplenote/src/main/java/com/automattic/simplenote/di/DataModule.kt	(revision 31029a7742cf2529eed6e65e50fb7048664c748c)
+++ Simplenote/src/main/java/com/automattic/simplenote/di/DataModule.kt	(date 1628528283397)
@@ -7,6 +7,7 @@
 import com.automattic.simplenote.repositories.SimperiumTagsRepository
 import com.automattic.simplenote.repositories.TagsRepository
 import com.simperium.client.Bucket
+import dagger.Binds
 import dagger.Module
 import dagger.Provides
 import dagger.hilt.InstallIn
@@ -15,15 +16,15 @@
 
 @Module
 @InstallIn(SingletonComponent::class)
-object DataModule {
-
-    @Provides
-    fun providesTagsBucket(@ApplicationContext appContext: Context): Bucket<Tag> = (appContext as Simplenote).tagsBucket
+abstract class DataModule {
+    companion object {
+        @Provides
+        fun providesTagsBucket(@ApplicationContext appContext: Context): Bucket<Tag> = (appContext as Simplenote).tagsBucket
 
-    @Provides
-    fun providesNotesBucket(@ApplicationContext appContext: Context): Bucket<Note> = (appContext as Simplenote).notesBucket
+        @Provides
+        fun providesNotesBucket(@ApplicationContext appContext: Context): Bucket<Note> = (appContext as Simplenote).notesBucket
+    }
 
-    @Provides
-    fun providesTagsRepository(tagsBucket: Bucket<Tag>, notesBucket: Bucket<Note>): TagsRepository =
-        SimperiumTagsRepository(tagsBucket, notesBucket)
-}
\ No newline at end of file
+    @Binds
+    abstract fun bindsTagsRepository(repository: SimperiumTagsRepository): TagsRepository
+}
Index: Simplenote/src/main/java/com/automattic/simplenote/repositories/SimperiumTagsRepository.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Simplenote/src/main/java/com/automattic/simplenote/repositories/SimperiumTagsRepository.kt	(revision 31029a7742cf2529eed6e65e50fb7048664c748c)
+++ Simplenote/src/main/java/com/automattic/simplenote/repositories/SimperiumTagsRepository.kt	(date 1628528399499)
@@ -17,8 +17,9 @@
 import kotlinx.coroutines.flow.callbackFlow
 import kotlinx.coroutines.flow.flowOn
 import kotlinx.coroutines.withContext
+import javax.inject.Inject
 
-class SimperiumTagsRepository(
+class SimperiumTagsRepository @Inject constructor(
         private val tagsBucket: Bucket<Tag>,
         private val notesBucket: Bucket<Note>
 ) : TagsRepository {
```

Reply:
That is a great patch @hichamboushaba! I have applied it in ac26d345b2a5438e69b8b8dd6cda827b467a4070.